File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Submit_product_review_026_Secure.java
import java.util.ArrayList;
import java.util.List;

public class Submit_product_review_026_Secure {

    static class Review {
        int rating;
        String review;

        Review(int rating, String review) {
            this.rating = rating;
            this.review = review;
        }
    }

    static List<Review> reviews = new ArrayList<>();

    public static String submitReview(
            int rating,
            String review) {

        if (review == null)
            return "Review text is required.";

        review = review.trim();

        if (review.isEmpty())
            return "Review text is required.";

        if (rating < 1 || rating > 5)
            return "Rating must be between 1 and 5.";

        reviews.add(new Review(rating, review));

        return "Review submitted successfully.";
    }

    public static void main(String[] args) {

        System.out.println(submitReview(
                5,
                "Highly recommended."
        ));
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/Submit_product_review_026_Secure.py
reviews = []


def submit_product_review(rating, review_text):
    """
    Securely submits a product review.
    """

    try:

        if review_text is None:
            return "Review text is required."

        review_text = review_text.strip()

        if not review_text:
            return "Review text is required."

        if not isinstance(rating, int):
            return "Invalid rating."

        if rating < 1 or rating > 5:
            return "Rating must be between 1 and 5."

        reviews.append({
            "rating": rating,
            "review": review_text
        })

        return "Product review submitted successfully."

    except Exception:
        return "Unable to submit review."


if __name__ == "__main__":
    print(submit_product_review(
        5,
        "Very good product."
    ))